packages/framework/src/Component/ClassExtension/MethodAnnotationsFactory.php

```diff
@@ -100,25 +98,54 @@ protected function isMethodImplementedInClass(string $methodName, ReflectionClas
      * @param \Roave\BetterReflection\Reflection\ReflectionMethod $reflectionMethod
      * @return string
      */
-    protected function getMethodParameterNames(ReflectionMethod $reflectionMethod): string
+    protected function getMethodParameterNamesWithTypes(ReflectionMethod $reflectionMethod): string
     {
-        $methodParameterNames = [];
+        $methodParameterNamesWithTypes = [];
         foreach ($reflectionMethod->getParameters() as $methodParameter) {
-            $methodParameterNames[] = '$' . $methodParameter->getName();
+            $methodParameterNamesWithTypes[] = sprintf(
+                '%s $%s',
+                $this->annotationsReplacer->replaceInParameterType($methodParameter),
```

Comment:
Just curious, what if a parameter does not have a type? This should never happen, right?

Reply:
evil evil question! Well, it SHOULD never happen, but who knows? :man_shrugging: When there is no type specified in the parameter annotation, then I got an error when trying to use the command :sob: tested using test:
```
1) Tests\FrameworkBundle\Unit\Component\ClassExtension\MethodAnnotationsFactoryTest::testGetProjectClassNecessaryMethodAnnotationsLines
InvalidArgumentException: Attempted to resolve "" but it appears to be empty

/var/www/html/vendor/phpdocumentor/type-resolver/src/TypeResolver.php:99
/var/www/html/vendor/roave/better-reflection/src/TypesFinder/ResolveTypes.php:32
/var/www/html/vendor/roave/better-reflection/src/TypesFinder/FindParameterType.php:62
/var/www/html/vendor/roave/better-reflection/src/Reflection/ReflectionParameter.php:355
/var/www/html/vendor/roave/better-reflection/src/Reflection/ReflectionParameter.php:336
/var/www/html/packages/framework/src/Component/ClassExtension/MethodAnnotationsFactory.php:147
/var/www/html/packages/framework/src/Component/ClassExtension/MethodAnnotationsFactory.php:68
/var/www/html/packages/framework/src/Component/ClassExtension/MethodAnnotationsFactory.php:46
/var/www/html/packages/framework/tests/Unit/Component/ClassExtension/MethodAnnotationsFactoryTest.php:76

```

The missing Type is reported by PHPStan as well, however...I do know, it seems like I should address this issue :grimacing:

Reply:
I was thinking about try-catch for that, and just silently skip such scenarios, but maybe it is absolutely ok not to solve this at all. In the end, when a developer starts debugging the error message, he should quickly find out that he has something terribly wrong in his code and should be able to find the issue quickly.

Reply:
I have the same feeling that something should be done in this matter

Reply:
Try-catch and show warning? Is it possible and not too much work?

Reply:
we discussed this - end-users should never encounter this issue - it would affect them only in a situation when a framework class would have badly formatted annotation and this can not happen as it is guarded by phpstan :grin: 
For code like this:
```
/**
 * @param $productData
 */
```
you get phpstan error
```
PHPDoc tag @param has invalid value
```